src/Raven.Server/Utils/TypeConverter.cs

```diff
@@ -390,42 +391,57 @@ internal static IEnumerable<object> Flatten(IEnumerable items)
             }
         }
 
+        private static readonly Type[] DynamicRules = new[]
+        {
+            typeof(DynamicNullObject), typeof(IDynamicMetaObjectProvider), typeof(string),
+            typeof(LazyCompressedStringValue), typeof(LazyStringValue), typeof(BlittableJsonReaderObject),
+            typeof(BlittableJsonReaderArray)
+        };
+
+        private enum DynamicRuleTypeLocation : int
+        {
+            DynamicNullObject = 0,
+            DynamicMetaObjectProvider = 1,
+            String = 2,
+            LazyCompressedStringValue = 3,
+            LazyStringValue = 4,
+            BlittableJsonReaderObject = 5,
+            BlittableJsonReaderArray = 6,
+        }
+
         public static dynamic ToDynamicType(object value)
         {
             if (value == null)
                 return DynamicNullObject.ExplicitNull;
 
-            if (value is DynamicNullObject)
-                return value;
+            Type objectType = value.GetType();
 
-            if (value is IDynamicMetaObjectProvider)
+            if (objectType == DynamicRules[(int)DynamicRuleTypeLocation.DynamicNullObject] ||
```

Comment:
I thought that `type == `typeof(Foo)` is the cheapest way to do that, no?
Turned into direct constant by the JIT?

Reply:
Apparently, it does not where there are so many of them. I have a few tests to perform yet on it, but this is why I believe this is faster: 
![image](https://user-images.githubusercontent.com/1045411/110461754-96f16780-80ae-11eb-9822-18b58d276dc4.png)
As you can see, if you are jumping around to get the object type through the pointers table every time; if we can skip that (which we did) we can win a lot.